packages/composites/proof-of-reserves/src/utils/protocol.ts

```diff
@@ -41,11 +45,36 @@ export const runProtocolAdapter = async (
   return callAdapter(execute, context, next, '_onProtocol')
 }
 
-const listAdapter = (jobRunID: string, data: { addresses: string[] }) => {
+const listAdapter = (
+  jobRunID: string,
+  data: { addresses: AddressList; chainId?: string; network?: string },
+) => {
+  console.log('LIST ADAPTER')
+  console.log(data)
   if (!('addresses' in data)) {
     throw Error(`Missing "addresses" in request data`)
   }
-
-  const result = data.addresses.map((address: string) => ({ address }))
-  return Requester.success(jobRunID, { data: { result } })
+  if (typeof data.addresses[0] === 'string') {
+    const result = (data.addresses as string[]).map((address: string) => {
+      return { address, network: data.network, chainId: data.chainId }
+    })
```

Comment:
Nbd but I like to remove the `return` keyword when possible
```suggestion
    const result = (data.addresses as string[]).map((address: string) => (
      { address, network: data.network, chainId: data.chainId }
    ))
```